src/adagrams.js

```diff
@@ -1,15 +1,175 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  static drawLetters = () => {
+    // Implement this method for wave 1
+    const LETTER_POOL = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    const letters = [];
+    for (const key in LETTER_POOL) {
+      for (let i = 0; i < LETTER_POOL[key]; i++) {
+        letters.push(key);
+      }
+    }
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
-};
+    const shuffled = letters.sort(() => 0.5 - Math.random());
+    let userHand = shuffled.slice(0, 10);
+    // console.log(userHand);
+    return userHand;
+  };
+
+
+  static usesAvailableLetters = (input, lettersInHand) => {
+    // Implement this method for wave 2
+    const itemCounter = (array, item) => {
+      let counter = 0;
+      array.flat(Infinity).forEach((x) => {
+        if (x == item) {
+          counter++;
+        }
+      });
+      return counter;
+    };
+
+    for (let letter of input) {
+      if (!lettersInHand.includes(letter)) {
+        return false;
+      } else {
+        if (
+          itemCounter(input.split(""), letter) >
+          itemCounter(lettersInHand, letter)
```

Comment:
Neat way to handle this!  One small optimization I suggest is to store the result of input.split("") in a helper variable before the for loop, so it only needs to be called once, rather than multiple times.  The itemCounter function does not change the array that's passed to it, so there's no need to re-create the split every time.